controllers/idle_handlers.go

```diff
@@ -18,16 +18,49 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"os"
 
 	ranv1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+func getOSName(seedImageVersion string) string {
+	return fmt.Sprintf("rhcos_%s", seedImageVersion)
+}
+
+func (r *ImageBasedUpgradeReconciler) removeDeployment(osName string) {
+	status, err := r.RpmOSTreeClient.QueryStatus()
+	if err != nil {
+		r.Log.Error(err, "Failed to query rpm-ostree")
+		return
+	}
+	for index, deployment := range status.Deployments {
+		if deployment.OSName == osName {
+			ostreeUndeployCommand := fmt.Sprintf("ostree admin undeploy %d", index)
+			utils.ExecuteChrootCmd(utils.Host, ostreeUndeployCommand)
+			return
+		}
+	}
+	r.Log.Info("Unable to find ostree deployment", "osName", osName)
+}
+
 //nolint:unparam
 func (r *ImageBasedUpgradeReconciler) handleAbort(ctx context.Context, ibu *ranv1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+	OSName := getOSName(ibu.Spec.SeedImageRef.Version)
+	r.removeDeployment(OSName)
 
-	// TODO actual steps
-	// If succeeds, return doNotRequeue
+	sysrootDeployPath := fmt.Sprintf("/sysroot/ostree/deploy/%s", OSName)
+	if _, err := os.Stat(sysrootDeployPath); err != nil {
+		removeSysrootFilesCommand := fmt.Sprintf("unshare -m /bin/sh -c \"mount -o remount,rw /sysroot && rm -rf %s\"", sysrootDeployPath)
+		utils.ExecuteChrootCmd(utils.Host, removeSysrootFilesCommand)
```

Comment:
As mentioned in my previous comment, the execute function needs to return err and the handler should decide what to do in different cases, either retry by itself or let it reconcile again by returning err or declare abort failure etc.

Reply:
do you mean change the execute functions in this PR?

Reply:
We can leave the actual error handling logic for later. But we should correct what the function returns before it's getting used everywhere.

Reply:
updated